### PR TITLE
Changelog for 0.5.0.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Final changelog. Quite a lot has changed in ~1 year and > 900 commits, so it's hard to be comprehensive here!

Do you see anything missing @petrelharp, @ashander?

### CODE REVIEW

**Comment by ashander** [comment]
* I guess we'll have a few sentences at the top mentioning the new tables API
* there are breaking changes to the hdf5 format, of course. but maybe that's not needed due to `msp upgrade`?
* possibly note the various `get_*` are deprecated

File: CHANGELOG.rst [hunk 1]
```diff
@@ -19,8 +19,76 @@
   ``mutation.index`` will still work through deprecated aliases,
   but new code should use a more cite-centric approach.
 
+- The ``TreeSequence.diffs()`` method no longer works. Please use
+  the ``TreeSequence.edge_diffs()`` method instead.
 
-**Deprecated aliases**:
+- TreeSequence.get_num_records() no longer works. Any code using
+  this or the records() iterator should be recast to work with
```
Comment (ashander) [inline CHANGELOG.rst]: probably not an issue but: maybe s/recast/rewritten/ as casting has that other meaning in CS

File: CHANGELOG.rst [hunk 2]
```diff
@@ -19,8 +19,76 @@
...
+  ``tree.get_total_branch_length()`` with ``tree.total_branch_length``.
+  The old forms have been maintained as deprecated aliases. (#64)
+
+- Support for a much more general class of tree sequence topologies.
+  For example, trees with multiple roots are fully supported.
+
+- Substantially generalised mutation model. Mutations now occur at
+  specific sites, which can be associated with zero to many mutations.
+  Each site has an ancestral state (any character string) and
+  each mutation a derived_state (any character string).
```
Comment (ashander) [inline CHANGELOG.rst]: either remove _ or put a similar one in ancestral_state

File: CHANGELOG.rst [hunk 3]
```diff
@@ -19,8 +19,76 @@
...
+  site, and genotypes are indexes into this array. This is both
+  consistent with existing usage, and generalisable to any character
+  strings used as mutational states.
+
+- Add the formal concept of a sample, and distinguished from 'leaves'.
+  Change ``tracked_leaves``, etc. to ``tracked_samples`` (#225).
+  Also rename ``sample_size`` to ``num_samples`` for consistency (#227).
+
+- Efficient interchange of tree sequence data using the new Tables
+  API.
```
Comment (ashander) [inline CHANGELOG.rst]: Just to give folks a handle (though searching the docs should work) maybe add:

```
This consists of several classes, e.g., ``NodeTable`` and the methods ``load_tables`` and ``load_text``.
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -19,8 +19,76 @@
...
+
+- Newick output is not supported for more general trees. (#117)
+
+- The ``genotype_matrix`` method allows efficient access to the
+  full genotype matrix. (#306)
+
+- The variants iterator no longer uses a single buffer for
+  genotype data, removing a common source of error (#253).
+
+- Unicode and ASCII output formats for SparseTree.draw().
```
Comment (ashander) [inline CHANGELOG.rst]: for consistency add `` around this method here and below

File: CHANGELOG.rst [hunk 5]
```diff
@@ -19,8 +19,76 @@
...
+
+- Substantially generalised mutation model. Mutations now occur at
+  specific sites, which can be associated with zero to many mutations.
+  Each site has an ancestral state (any character string) and
+  each mutation a derived_state (any character string).
+
+- The variants() method now returns a list of alleles for each
+  site, and genotypes are indexes into this array. This is both
+  consistent with existing usage, and generalisable to any character
+  strings used as mutational states.
```
Comment (ashander) [inline CHANGELOG.rst]: I find the last clause a bit unclear. But I might misunderstand something. Would subbing this for L47-L48 still say the same thing? 

```
consistent with existing usage and works with the newly generalised mutation model,
which allows arbitrary strings of characters as mutational states.
```

If so I find this wording clearer.

**Comment by jeromekelleher** [comment]
Thanks for the review @ashander, this was really helpful. I've pushed an update with your edits (plus a few more).

File: CHANGELOG.rst [hunk 6]
```diff
@@ -19,8 +22,81 @@
   ``mutation.index`` will still work through deprecated aliases,
   but new code should use a more cite-centric approach.
```
Comment (petrelharp) [inline CHANGELOG.rst]: I don't know what "cite-centric" means.

**Comment by petrelharp** [comment]
Maybe a pointer to the substantially updated docs?

There isn't any mention of the new statistics. Maybe that's OK since we plan to really release these later; but could mention these are here, in beta:
```
Initial support for a broad class of site-based statistics, including Patterson's f-statistics, has been added, through the `SiteStatCalculator`, and its branch length analog, `BranchLengthStatCalculator`.  The interface is still in development, and may change.
```

**Comment by jeromekelleher** [comment]
Thanks @petrelharp, I've updated with these.

The only reason for delaying the actual release now is that I was hoping to have a conda package on conda-forge ready for the 0.5.0 release. I want to migrate the current conda package from biopython to conda-forge so that we can have easy windows support. If that's done, then we can update the installation docs accordingly.

Getting the package onto conda-forge will take some time though, as the admins are *very* busy.

**Comment by ashander** [comment]
The only reason for delaying the actual release now is that I was hoping to
> have a conda package on conda-forge ready for the 0.5.0 release. I want to
> migrate the current conda package from biopython to conda-forge so that we
> can have easy windows support. If that's done, then we can update the
> installation docs accordingly.
>
> Getting the package onto conda-forge will take some time though, as the
> admins are *very* busy.
>

Seems worth the wait to have clear instructions and easy windows support

**Comment by jeromekelleher** [comment]
I'm going to merge this now as we all seem reasonably happy with it.

**Pull request merged**